src/wallet/wallet.cpp

```diff
@@ -3261,9 +3261,9 @@ std::map<CTxDestination, CAmount> CWallet::GetAddressBalances()
 
     {
         LOCK(cs_wallet);
-        BOOST_FOREACH(PAIRTYPE(uint256, CWalletTx) walletEntry, mapWallet)
+        for (const auto& walletEntry : mapWallet)
         {
-            CWalletTx *pcoin = &walletEntry.second;
+            const CWalletTx *pcoin = &walletEntry.second;
```

Comment:
> personal preference, just in case you agree: I'd use const CWalletTX& pcoin = walletEntry.second; instead.

I'd be fine with this change, but I'm not so motivated to implement it myself, and kind of think it would sidetrack the pr which right now is very simple.